nucliadb_node/src/shards/providers/unbounded_cache/async_unbounded_reader.rs

```diff
@@ -38,16 +39,16 @@ pub struct AsyncUnboundedShardReaderCache {
 }
 
 impl AsyncUnboundedShardReaderCache {
-    pub fn new(shards_path: PathBuf) -> Self {
+    pub fn new(settings: Arc<Settings>) -> Self {
```

Comment:
Doubt: why is it better to propagate all `Settings` object instead of passing the required parameters?

Reply:
it's probably not better, I was just going for some consistency in usage

I can change it to passing all the settings if you feel strongly

I think I was thinking that passing settings would prevent needing to continually churn around wiring different params up

but I really don't care either way

Reply:
No, it's fine, I was just wondering. I agree it makes the code more consistent although we are passing unneeded information